reg_tests/CTestList.cmake

```diff
@@ -2,79 +2,111 @@
 # Functions for adding tests / Categories of tests
 #=============================================================================
 
+macro(setup_test testname np)
+  set(TEST_WORKING_DIR "${CMAKE_CURRENT_BINARY_DIR}/test_files/${testname}")
+  file(MAKE_DIRECTORY ${TEST_WORKING_DIR})
+  if(NALU_WIND_SAVE_GOLDS)
+    file(MAKE_DIRECTORY ${SAVED_GOLDS_DIR}/${testname})
+    set(SAVE_GOLDS_COMMAND "--save-norm-file ${SAVED_GOLDS_DIR}/${testname}/${testname}.norm.gold")
+    set(SAVE_GOLDS_COMMAND_NP "--save-norm-file ${SAVED_GOLDS_DIR}/${testname}/${testname}Np${np}.norm.gold")
+  endif()
+  set(MPI_PREAMBLE "${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${np} ${MPIEXEC_PREFLAGS}")
+  set(MPI_COMMAND "${MPI_PREAMBLE} ${CMAKE_BINARY_DIR}/${nalu_ex_name} ${MPIEXEC_POSTFLAGS}")
+  set(INPUT_FILE_BASE "${CMAKE_CURRENT_SOURCE_DIR}/test_files/${testname}/${testname}")
+  set(INPUT_FILE "${INPUT_FILE_BASE}.yaml")
+  set(INPUT_FILE_RST "${INPUT_FILE_BASE}_rst.yaml")
+  set(INPUT_FILE_R0 "${INPUT_FILE_BASE}_R0.yaml")
+  set(INPUT_FILE_R1 "${INPUT_FILE_BASE}_R1.yaml")
+  set(OUTPUT_FILE "${testname}.log")
+  set(OUTPUT_FILE_RST "${testname}_rst.log")
+  set(OUTPUT_FILE_R0 "${testname}_R0.log")
+  set(OUTPUT_FILE_R1 "${testname}_R0.log")
+  set(OUTPUT_FILE_NP "${testname}Np${np}.log")
+  set(RUN_COMMAND "${MPI_COMMAND} -i ${INPUT_FILE} -o ${OUTPUT_FILE}")
+  set(RUN_COMMAND_RST "${MPI_COMMAND} -i ${INPUT_FILE_RST} -o ${OUTPUT_FILE_RST}")
+  set(RUN_COMMAND_R0 "${MPI_COMMAND} -i ${INPUT_FILE_R0} -o ${OUTPUT_FILE_R0}")
+  set(RUN_COMMAND_R1 " && ${MPI_COMMAND} -i ${INPUT_FILE_R1} -o ${OUTPUT_FILE_R1}")
+  set(RUN_COMMAND_NP "${MPI_COMMAND} -i ${INPUT_FILE} -o ${OUTPUT_FILE_NP}")
+  set(COMPARE_GOLDS_COMMAND_BASE " && ${CMAKE_CURRENT_SOURCE_DIR}/check_norms.py --abs-tol ${ABS_TOLERANCE} --rel-tol ${REL_TOLERANCE}")
+  set(COMPARE_GOLDS_COMMAND "${COMPARE_GOLDS_COMMAND_BASE} ${testname} ${NALU_WIND_REFERENCE_GOLDS_DIR}/${testname}/${testname}.norm.gold ${SAVE_GOLDS_COMMAND}")
```

Comment:
How does one indicate to CTest that they want to compare against golds saved in a custom gold norms dir?